### PR TITLE
Update book with grammar and link fixes

### PR DESCRIPTION
**Pull request by Hoverth**
Hi there, I was reading through the Asterinas book and found some grammar errors and a few broken links, so I've tried to fix them.

I also propose a bit of a solution for #1014, with a placeholder name (something that I figured OSTD stood for) for the Asterinas OSTD index in the book

### CODE REVIEW

File: docs/src/osdk/guide/run-project.md [hunk 1]
```diff
@@ -76,12 +76,12 @@ cargo osdk test foo
...
 launch the kernel.
 The following documentations provide details on
 all the available options:
 
-- [build options](../reference/commands/build.md)
-- [run options](../reference/commands/run.md)
-- [test options](../reference/commands/test.md)
+- [`build` options](../reference/commands/build.md)
+- [`run` options](../reference/commands/run.md)
+- [`test` options](../reference/commands/test.md)
```
Comment (Hoverth) [inline docs/src/osdk/guide/run-project.md:87]: I'm not sure how this would look like when formatted into the book, any advice?

Comment (tatetian) [reply docs/src/osdk/guide/run-project.md:87]: I think the changes will be formatted just fine. If you want to preview the rendered book, run these commands

```
cd docs
mdbook serve --open
```

**Review by tatetian — APPROVED** [review]
Thanks for helping with the docs! The changes look great. I am happy to accept this PR.

**Comment by tatetian** [comment]
@Hoverth Could you rebase to the latest main branch to resolve the conflict?

It seems that the last commit is a merge commit. Asterinas wants to maintain a linear history of Git commits; so no merge commits are allowed. So you need to _rebase_ your branch to the latest main branch, not merge your branch with it. Thanks.

**Comment by Hoverth** [comment]
@tatetian 
Does that fix the merge commit issue?

**Comment by tatetian** [comment]
> @tatetian Does that fix the merge commit issue?

Yes, the merge commit is removed. But now a new problem emerges: there are now some intermediate commits (“resolve rebasing ...") that should NOT be persisted in the commit history of the main branch. Since all changes in this PR is about documentation, I think it is acceptable to _squash_ them into one commit.

**Comment by Hoverth** [comment]
There, that should squash all the intermediate commits. Thank you for your patience.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**